### DISCUSSION TITLE
How to handle a mis-spelled word in an interface

### DISCUSSION THREAD

**Discussion by ckipp01** [sbt/sbt#7063] [Q&A]
So this is a bit embarrassing. I messed this up when I sent this in.

https://github.com/sbt/sbt/blob/4841e85af0676ee0a659765dcc4dbb4076607d35/internal/util-interface/src/main/java/xsbti/Problem.java#L48-L50

```diff
-  default List<DiagnosticRelatedInformation> diagnosticRelatedInforamation() {
+  default List<DiagnosticRelatedInformation> diagnosticRelatedInformation() {
    return Collections.emptyList();
  }
```

So how should I handle this?

1. Deprecate the old one and add a new correct one?
2. Just correct it (I added this to add in Dotty, and luckily I added the `DiagnosticCode`, but not `DiagnosticRelatedInformation` yet. So in theory, nothing is using this yet as nothing is reporting anything here. So again...  in theory it shouldn't be an issue, but it's technically breaking compat if I do this
3. Something else?

**Comment by eed3si9n** [comment]
Yea, I've noticed it as well after releasing Zinc artifact. My suggestion is keep rolling with it since it only affects tooling authors.

**Reply by ckipp01** [reply]
Evidence of my mistake... forever there.

**Reply by adpi2** [reply]
New contributors will stumble upon this typo and will probably try to correct it, for better or worse. So I suggest we do `1. Deprecate and add a new method` now, or at least we should add a comment.

**Reply by eed3si9n** [reply]
Yea. I guess we can fix it before it spreads too much.

**Reply by eed3si9n** [reply]
Note there is a matching misspelling in parameter name

https://github.com/sbt/sbt/blob/1b3f9a6357e12ea4db66d1f1dac18fedec81747d/internal/util-logging/src/main/scala/sbt/util/InterfaceUtil.scala#L140

I had also copied the mispelling intentionally to protobuf scheme for consistency:
https://github.com/sbt/zinc/blob/7d91a6974d12d35a80f7e8f524c46c94843df6f8/internal/zinc-persist-core/src/main/protobuf/schema.proto#L99

Since I don't want to break things, this fix will need to be done some time after 1.8.0.
